src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -489,17 +490,27 @@ private string FormatProcessOutputs((string StandardOutput, string StandardError
 
             // Allow the process time to send messages to the above delegates
             // if the process exits very quickly
-            System.Threading.Thread.Sleep(1000);
+            Thread.Sleep(1000);
 
-            // If we allow any timespan to be max value, it could cause unusual behavior, force max-int max (still a LOT)
-            // (int.MaxValue ms is about 24 days).  Large values are effectively timeouts for the outer harness
-            if (!p.WaitForExit((int)Math.Min(timeOut.TotalMilliseconds, int.MaxValue)))
+            bool finishedNormally = p.WaitForExit((int)Math.Min(timeOut.TotalMilliseconds, int.MaxValue));
+
+            // Lock the stringbuilders used as rarely this can cause concurrency issues
+            // resulting in "Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'chunkLength')"
+            lock (standardOut)

```

Comment:
Approving, just a nit: you can do:
```csharp
lock (standardOut)
lock (standardErr)
{
}
```
That way we reduce a little the indentation, but that is only useful for people like me that use terminal editors :)